api/src/trace/NoopTracer.ts

```diff
@@ -99,11 +99,15 @@ export class NoopTracer implements Tracer {
   }
 }
 
-function isSpanContext(spanContext: any): spanContext is SpanContext {
+function isSpanContext(spanContext: unknown): spanContext is SpanContext {
   return (
+    spanContext !== null &&
     typeof spanContext === 'object' &&
+    'spanId' in spanContext &&
     typeof spanContext['spanId'] === 'string' &&
+    'traceId' in spanContext &&
     typeof spanContext['traceId'] === 'string' &&
+    'traceFlags' in spanContext &&
     typeof spanContext['traceFlags'] === 'number'
   );
```

Comment:
This imposes a very slight runtime cost from the additional check. If that is undesirable here, there are plenty of alternative factoring that would accomplish the same without runtime code, this is just the most readable IMO. 

Alternatively, if we trust the types, this function can be deleted altogether. It's only ever called from L47; the input came from `getSpanContext()`, which is typed to return `SpanContext | undefined`. If that is accurate (looks like it to me), then all we really needed to check is `!== undefined` and it can be done inline.

Reply:
I think I prefer the option to remove this function altogether. The only way for this to not be true is for someone to really try very hard to dig into unexposed and unpublished internals. IIRC the original motivation of this function was to differentiate between `Span` and `SpanContext` but it seems like that's no longer relevant.